File: 2017_04_xfmoulet.py
n1 = n2 = 0
for line in open("2017_04.data") :
	words = line.split()
	sorted_words = [''.join(sorted(w)) for w in words]

	if len(words) == len(set(words)) : n1 += 1
	if len(sorted_words) == len(set(sorted_words)) : n2 += 1

print(n1,n2)

File: 2017/2017_04_xfmoulet.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func all_unique(arr []string) bool {
	uniques := map[string]bool{}
	for _, w := range arr {
		_, present := uniques[w]
		if present {
			return false
		} else {
			uniques[w] = true
		}
	}
	return true
}

func main() {
	fmt.Println("hello")
	file, err := os.Open("2017_04.data")
	if err != nil {
		fmt.Println("Error reading file")
		return
	}
	defer file.Close()

	// read file line by line
	nlines := 0
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		words := strings.Split(scanner.Text(), " ")
		is_unique := all_unique(words)
		fmt.Println(words, is_unique)
		if is_unique {
			nlines += 1
		}
	}
	fmt.Println(nlines)
}
